src/Tools/AI Test Toolkit/README.md

```diff
@@ -0,0 +1,91 @@
+    1. If the test output is set, it must be generated for all the tests which needs to be evaluated externally
+1. Results are logged in AI 'Log Entries'
+
+### Inspect the tesults
+1. Open Log Entries to result of each execution
+1. Download the test output which generates the `.jsonl` file or export the logs to Excel
+1. You can also use the API (page 149038 "AIT Log Entry API") to get the result for a suite
+1. Open AL Test Tool and switch to the created test suite to execute each test manually
+
+
+## Writing data-driven AI tests
+
+### Defining test codeunit
+A data-driven AI test is defined like any normal AL test, except that it:
+1. Is executed through the BC Copilot Test Toolkit
+2. Utilizes the `AIT Test Context` codeunit to get input for the test
+3. _(Optional)_ Sets the output using the `AIT Test Context` codeunit
+
+See the `AIT Test Context` for full API.
+
+#### Example
+An example test for an AI feature that returns an integer.
+
+```
+    [Test]
+    procedure TestCopilotFeature()
+    var
+        AITestContext: Codeunit "AIT Test Context";
+        Question: Text;
+        Output: Integer;
+        ExpectedOutput: Integer;
+    begin
+        // [Scenario] AI Test
+
+        // Get the input from the dataset
+        TestInput := AITestContext.GetQuestion().ValueAsText();
+
+        // Call the LLM to get an output
+        Output := CopilotFeature.CallLLM(Question);
+
+        // Get the expected output from the dataset
+        ExpectedOutput := AITestContext.GetExpectedData().ValueAsInteger();
+
+        // Assert the result
+        Assert.AreEqual(ExpectedOutput, Output, '');
+    end;
+```
+In this example
+1. This test procedure will be called with each input from the dataset
+1. We get the `question` and `expected_data` from the input dataset using `AITestContext.GetQuestion()` and `AITestContext.GetExpectedData()` respectively. 
+1. Alternatively, we could use `AITestContext.GetInput()` and get the line as `JSON`. 
+
+
+### Defining Datasets
+Datasets are provided as `.jsonl` files where each line represents a test case.
+
+There's no fixed structure required for each line, but using certain formatting will allow easier access to data and name and description definition.
+
+See the `AIT Test Context` for full API.
+
+#### Example
+
+```
+{"name": "Test01", "question": "A question", "expected_data": 5}
+{"name": "Test02", "question": "A second question", "expected_data": 2}
+{"name": "Test03", "question": "A third question", "expected_data": 2}
+```
+
+In this example
+1. Since we're setting `name` for each line, that will be used in the BC Copilot Test Toolkit when uploading the dataset.
+1. Since we're setting `question` for each line, we can use `AITestContext.GetQuestion()` in the test codeunit, to get the question directly.
+1. Since we're setting `expected_data` for each line, we can use the `AITestContext.GetExpectedData()` in the test codeunit, to get the question directly..
```

Comment:
```suggestion
1. Since we're setting `expected_data` for each line, we can use the `AITestContext.GetExpectedData()` in the test codeunit, to get the question directly.
```

Reply:
Thanks, fixed (no bullet points end with a period)